src/Moryx.Resources.Management/Resources/ResourceManager.cs

```diff
@@ -150,8 +196,22 @@ private void LoadResources(ICollection<ResourceEntityAccessor> allResources)
             Parallel.ForEach(allResources, LinkReferences);
 
             // Register events after all links were set
-            foreach (var resourceWrapper in Graph.GetAll())
-                RegisterEvents(resourceWrapper.Target);
+            foreach (var resource in Graph.GetAll())
+                RegisterEvents(resource);
+        }
+
+        /// <summary>
+        /// Handles the resource started by adding it to the list of running resources and removes it from failed resources
+        /// </summary>
+        /// <param name="resource">Started resource</param>
+        private void HandleResourceStarted(IResource resource)
+        {
+            lock (_failedResources)
+            {
+                //check if the resource exist in failed resources
+                if (_failedResources.FirstOrDefault(x => x.Id == resource.Id) is not null)
```

Comment:
If is not necessary. Just call `Remove`, it returns false if no entry was found.

But in fact: We talked about reducing the life cylce, so a resource can not be failed before initialize and it will never reach Start if it failed in Initialize.

We only need the failed collection when responding to requests on the facade.